ignite/handlers/checkpoint.py

```diff
@@ -386,6 +388,7 @@ def __init__(
                     "".format(matched, dirname)
                 )
 
+    @one_rank_only()
```

Comment:
We need to rework `_save` method here:
According to the code of `xm.save`, it does barrier in the end of the method: https://github.com/pytorch/xla/blob/master/torch_xla/core/xla_model.py#L632
So, if we do one rank only, it will get stuck.

Reply:
Ok so we can't decorate and we have to do conditional branches (already done)
```python
def _save(self, checkpoint: Mapping, filename: str):
    if idist.has_xla_support:
        import torch_xla.core.xla_model as xm

        xm.save(checkpoint, filename)
    elif idist.get_rank() == 0:
        torch.save(checkpoint, filename)
```
The `atomic` mode should be checked with multinode `xla` (work with a file per node? file per process?).